idaes/generic_models/properties/core/eos/enrtl.py

```diff
@@ -109,70 +158,171 @@ def rule_X(b, j):
+                return Expression.Skip
+            elif j in b.params.cation_set or j in b.params.anion_set:
+                return (x[j]*cobj(b, j).config.charge)
+            else:
+                return 0
+
+        b.add_component(
+            pname+"_X_ref",
+            Expression(b.params.true_species_set,
+                       rule=rule_X_ref,
+                       doc="Charge x mole fraction term at reference state"))
+
         def rule_Y(b, j):
-            cobj = b.params.get_component(j)
-            if cobj.config.charge < 0:
+            if cobj(b, j).config.charge < 0:
                 # Anion
                 dom = b.params.anion_set
             else:
                 dom = b.params.cation_set
 
             X = getattr(b, pname+"_X")
-            return X[j]/sum(X[i] for i in dom)
-
+            return X[j]/sum(X[i] for i in dom)  # Eqns 36 and 37
+        # Y is a charge ratio, and thus independent of x for symmetric state
+        # TODO: This may need to change for the unsymmetric state
         b.add_component(pname+"_Y",
                         Expression(b.params.ion_set,
                                    rule=rule_Y,
                                    doc="Charge composition"))
 
+        # ---------------------------------------------------------------------
+        # Long-range terms
+        # Average molar volume of solvent
+        def rule_vol_mol_solvent(b):  # Eqn 77
+            return (sum(b.mole_frac_phase_comp_true[pname, s] /
+                        get_method(b, "dens_mol_liq_comp", s)(
+                            b, cobj(b, s), b.temperature)
+                        for s in molecular_set) /
+                    sum(b.mole_frac_phase_comp_true[pname, s]
+                        for s in molecular_set))
+        b.add_component(pname+"_vol_mol_solvent",
+                        Expression(rule=rule_vol_mol_solvent,
+                                   doc="Mean molar volume of solvent"))
+
+        # Mean relative permitivity of solvent
+        def rule_eps_solvent(b):  # Eqn 78
+            return (sum(b.mole_frac_phase_comp_true[pname, s] *
+                        get_method(b, "relative_permittivity_liq_comp", s)(
+                            b, cobj(b, s), b.temperature) *
+                        b.params.get_component(s).mw
+                        for s in molecular_set) /
+                    sum(b.mole_frac_phase_comp_true[pname, s] *
+                        b.params.get_component(s).mw
+                        for s in molecular_set))
+        b.add_component(pname+"_relative_permittivity_solvent",
+                        Expression(
+                            rule=rule_eps_solvent,
+                            doc="Mean relative permittivity  of solvent"))
+
+        # Debye-Huckel parameter
+        def rule_A_DH(b):  # Eqn 61
+            # Note: Where the paper refers to the dielectric constant, it
+            # actually means the electric permittivity of the solver
```

Comment:
`solver` --> `solvent`

Reply:
Seeing as this PR has already been approved and is almost ready to merge, I'll fix this in the subsequent PR.